python/lsst/meas/algorithms/detection.py

```diff
@@ -366,11 +370,36 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True):
 
         fpSets = pipeBase.Struct(positive=None, negative=None)
 
+        if self.config.doTempLocalBackground:
+            # Subtract the local background from the smoothed image.
```

Comment:
But if that's true, then
```
add_background(smooth(subtract_background(image))) != smooth(image)
```
but think we still have:
```
add_background(subtract_background(smooth(image))) == smooth(image)
```
so I think then we *definitely* need to smooth before subtracting.